test/integ/zeroDimBuffer/src/zeroDimBuffer.cpp

```diff
@@ -0,0 +1,185 @@
+/* Copyright 2021 Andrea Bocci
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+// make sure the CPU_B_SEQ_T_SEQ backend is always available
+#ifndef ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED
+#    define ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED
+#endif // ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED
+#include <alpaka/alpaka.hpp>
+#include <alpaka/test/acc/TestAccs.hpp>
+
+#include <catch2/catch.hpp>
+
+#include <cstdint>
+#include <cstring>
+#include <iostream>
+
+// fill an trivial type with std::memset
+template<typename T>
+constexpr T memset_value(int c)
+{
+    T t;
+    std::memset(&t, c, sizeof(T));
+    return t;
+}
+
+//! check if asynchronous (queue-ordered) memory buffers are supported by the given Accelerator
+template<typename TAcc>
+static constexpr auto isAsyncBufferSupported() -> bool
+{
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+    if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevCudaRt>)
+    {
+        return (BOOST_LANG_CUDA >= BOOST_VERSION_NUMBER(11, 2, 0)) && (alpaka::Dim<TAcc>::value == 1);
+    }
+#endif // ALPAKA_ACC_GPU_CUDA_ENABLED
+
+#ifdef ALPAKA_ACC_GPU_HIP_ENABLED
+    if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevHipRt>)
+    {
+        return false;
+    }
+#endif // ALPAKA_ACC_GPU_HIP_ENABLED
+
+#ifdef ALPAKA_ACC_ANY_BT_OACC_ENABLED
+    if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevOacc>)
+    {
+        return false;
+    }
+#endif // ALPAKA_ACC_ANY_BT_OACC_ENABLED
+
+#ifdef ALPAKA_ACC_ANY_BT_OMP5_ENABLED
+    if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevOmp5>)
+    {
+        return false;
+    }
+#endif // ALPAKA_ACC_ANY_BT_OMP5_ENABLED
+
+    return true;
```

Comment:
The CI is saying that this statement is not reachable if `OpenAcc` is enabled. 
I suggest using `#if defined() ... #elif defined() ... #else ... #endif`

Reply:
Well, yes:
  - `#ifdef ALPAKA_ACC_ANY_BT_OACC_ENABLED` will be true, so the corresponding `if constexpr` is checked;
  - `if constexpr(std::is_same_v<alpaka::Dev<TAcc>, alpaka::DevOacc>)` will be true, so the corresponding `return false` is executed;
  - the `return true` at the end of the function is not reachable.

I would say that this is really by construction. How does one silence the corresponding warning or error ?

Reply:
Maye, not nice but the only way which coming up to my mind at the moment.

```C++
bool result = true;
# ifdef ACC1
if constexpr ...
  result = false;
#endif
# ifdef ACC2
if constexpr ...
  result = true;
#endif
return result;
```

Reply:
@fwyzard How do you solved the issue with the `unreachable code`? 
Somehow you solved it because the CI is not passing but I can not see why, is it because of line 12?

Reply:
I don't remember having made changes here - but in the meantime I did fix the OpenMP 5 and OpenAcc `memcpy` and `memset` implementations for zero-dimensional buffers.